app/bundles/LeadBundle/Helper/TokenHelper.php

```diff
@@ -95,13 +98,40 @@ private static function getTokenValue(array $lead, $alias, $defaultValue)
             $value = $lead['companies'][0][$alias];
         }
 
-        if ('true' === $defaultValue) {
-            $value = urlencode($value);
-
+        if ($value) {
+            switch ($defaultValue) {
+                case 'true':
+                    $value = urlencode($value);
+                    break;
+                case 'datetime':
+                case 'date':
+                case 'time':
+                    $dt   = new DateTimeHelper($value);
+                    $date = $dt->getDateTime()->format(
+                        (new ParamsLoaderHelper())->getParameters()['date_format_dateonly']
```

Comment:
@kuzmany can you please explain why are you loading the config files here again? The middleware trait is loading it as at that point Symfony is not booted. Why didn't you just add this helper to the config.php:

https://github.com/mautic/mautic/blob/staging/app/bundles/LeadBundle/Config/config.php#L881

And injected the CoreParametersHelper?

Reply:
It's static method? Can we use DI in static method? I don't think so, maybe I am wrong.

Reply:
I missed that it's in a static method. It should be rewritten to new service with dependency to CoreParametersHelper. Loading all the config files from file system every time the ParamsLoaderHelper is initialized will be expensive and slow.

Also I think that the `date_format_dateonly` is missing form local.php when you first install Mautic. It will be written there once you save the config form for the first time. So for those users this will cause a PHP error.

Reply:
Rewrite mean BC break, isn't it?
Or we can write new TokenReplacer.php with DI and let TokenHelper with depraced removed in 3.0?

Or do you have another idea?

Reply:
Yes, you get that right. Rewrite is not the right expression. We have to replace it with a new service and mark this class as deprecated.

Reply:
Ok. I can prepare it to 2.14.2? Or it's late at the moment and  focus this feature to 2.15?

Reply:
Try to submit the PR asap. I'll try to get it in 2.14.2.

Reply:
Need more time. I started work, but want create is as service with interface, abstract, tests for re-usage in another app/bundles

https://github.com/mautic/mautic/pull/6710

Reply:
I woke up today and finish it
https://github.com/mautic/mautic/pull/6710

It written very clean with unit test. But changes are across all Mautic core and channels.